### PR TITLE
feat(nuget): fetch sourceUrl from all v3 registries

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Try to fetch sourceUrl from nuspec file for all v3 registries

## Context:

Currently we only fetch it from official nuget v3 registry.
`sourceUrl` is required to group packages by `sourceUrlPrefixes`

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/nuget/index.spec.ts [hunk 1]
```diff
@@ -404,19 +409,30 @@ describe('datasource/nuget', () => {
...
+          200,
+          pkgListV3Registration
+            .replace(/"http:\/\/nunit\.org"/g, '""')
+            .replace('"published": "2012-10-23T15:37:48+00:00",', '')
+        )
+        .get('/v3-flatcontainer/nunit/3.12.0/nunit.nuspec')
+        .reply(
+          200,
+          pkgInfoV3FromNuget.replace('https://github.com/nunit/nunit', '')
+        );
```
Comment (JamieMagee) [inline lib/datasource/nuget/index.spec.ts:422]: Is it worthwhile to have a test checking that `sourUrl` is only set once? i.e. return pkgInfo with 2 different homepages, and make sure that only the first one is used?

File: lib/datasource/nuget/v3.ts [hunk 2]
```diff
@@ -165,26 +169,43 @@ export async function getReleases(
...
+      `Cannot obtain sourceUrl`
+    );
+    return dep;
   }
 
+  // istanbul ignore else: not easy testable
   if (homepage) {
+    // only assign if not assigned
+    dep.sourceUrl ??= homepage;
     dep.homepage = homepage;
```
Comment (JamieMagee) [inline lib/datasource/nuget/v3.ts:209]: There's potential for `homepage` and `sourceUrl` to be different?

```suggestion
    dep.homepage ??= homepage;
```

File: lib/datasource/nuget/index.spec.ts (hunk 1 above)
Comment (viceice) [reply lib/datasource/nuget/index.spec.ts:422]: This is just to make coverage happy. 😇

File: lib/datasource/nuget/v3.ts (hunk 2 above)
Comment (viceice) [reply lib/datasource/nuget/v3.ts]: Good catch 👍

File: lib/datasource/nuget/index.spec.ts (hunk 1 above)
Comment (JamieMagee) [reply lib/datasource/nuget/index.spec.ts:422]: Okay, no worries.

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**